src/engine/LocalVocab.h

```diff
@@ -111,11 +112,11 @@ class LocalVocab {
   // primary set of this `LocalVocab` remains unchanged.
   template <std::ranges::range R>
   void mergeWith(const R& vocabs) {
-    auto inserter = std::back_inserter(otherWordSets_);
     using std::views::filter;
     for (const auto& vocab : vocabs | filter(std::not_fn(&LocalVocab::empty))) {
-      std::ranges::copy(vocab.otherWordSets_, inserter);
-      *inserter = vocab.primaryWordSet_;
+      otherWordSets_.insert(vocab.otherWordSets_.begin(),
+                            vocab.otherWordSets_.end());
+      otherWordSets_.insert(vocab.primaryWordSet_);
```

Comment:
1. Do we really have no tests for this case ? Recursively merging local vocabs?  If so, please add one .

2. Please comment the `filter(notEmpty)` above with something like `"Note: Even though the `otherWordsSet_` is a hash set that filters out duplicates, we still manually filter out empty sets, because these typically don't compare equal to each other because of the `shared_ptr` semantics.